src/core/segment_buffers/implementations/text/html/html_text_segment_buffer.ts

```diff
@@ -182,76 +214,162 @@ export default class HTMLTextSourceBuffer
   }
 
   /**
-   * Append text tracks.
+   * Push segment on Subscription.
+   * @param {Object} infos
+   * @returns {Observable}
+   */
+  public pushChunk(infos : IPushChunkInfos<IHTMLTextTrackData>) : Observable<void> {
+    return observableDefer(() => {
+      const hasPushedData = this.pushDataSync(infos.data);
+      if (!hasPushedData) {
+        return observableOf(undefined);
+      }
+
+      if (infos.inventoryInfos !== null) {
+        this._segmentInventory.insertChunk(infos.inventoryInfos);
+      }
+      return observableOf(undefined);
+    });
+  }
+
+  /**
+   * Remove buffered data.
+   * @param {number} start - start position, in seconds
+   * @param {number} end - end position, in seconds
+   * @returns {Observable}
+   */
+  public removeBuffer(start : number, end : number) : Observable<void> {
+    return observableDefer(() => {
+      this.removeBufferSync(start, end);
+      return observableOf(undefined);
+    });
+  }
+
+  /**
+   * Indicate that every chunks from a Segment has been given to pushChunk so
+   * far.
+   * This will update our internal Segment inventory accordingly.
+   * The returned Observable will emit and complete successively once the whole
+   * segment has been pushed and this indication is acknowledged.
+   * @param {Object} infos
+   * @returns {Observable}
+   */
+  public endOfSegment(_infos : IEndOfSegmentInfos) : Observable<void> {
+    return observableDefer(() => {
+      this._segmentInventory.completeSegment(_infos);
+      return observableOf(undefined);
+    });
+  }
+
+  /**
+   * Returns the currently buffered data, in a TimeRanges object.
+   * @returns {TimeRanges}
+   */
+  public getBufferedRanges() : ManualTimeRanges {
+    return this._buffered;
+  }
+
+  public dispose() : void {
+    log.debug("HTSB: Disposing HTMLTextSegmentBuffer");
+    this._hideCurrentCue();
+    this._buffer.remove(0, Infinity);
+    this._buffered.remove(0, Infinity);
+    this._destroy$.next();
+    this._destroy$.complete();
+  }
+
+  /**
+   * Push the text track contained in `data` to the HTMLTextSegmentBuffer
+   * synchronously.
+   * Returns a boolean:
+   *   - `true` if text tracks have been added the the HTMLTextSegmentBuffer's
+   *     buffer after that segment has been added.
+   *   - `false` if no text tracks have been added the the
+   *     HTMLTextSegmentBuffer's buffer (e.g. empty text-track, incoherent times
+   *     etc.)
+   *
+   * /!\ This method won't add any data to the linked inventory.
+   * Please use the `pushChunk` method for most use-cases.
    * @param {Object} data
+   * @returns {boolean}
    */
-  _append(data : IHTMLTextTrackData) : void {
-    log.debug("HTSB: Appending new html text tracks", data);
+  public pushDataSync(data : IPushedChunkData<IHTMLTextTrackData>) : boolean {
```

Comment:
Seems not so usual in the RxPlayer to return a boolean to indicate whether it has succeeded or not.

Reply:
It's true.

I will look again to see if I can do something different

Reply:
Done. I renamed it `pushChunkSync` and it does the exact same thing than `pushChunk` without wrapping it in an Observable (needed by the TextTrackRenderer, to ensure that parsing subtitles is done synchronously.

Reply:
Good !